src/sentry/filestore/s3.py

```diff
@@ -229,7 +229,7 @@ class S3Boto3Storage(Storage):
     mode and supports streaming(buffering) data in chunks to S3
     when writing.
     """
-    connection_class = staticmethod(resource)
+    connection_class = staticmethod(Session)
```

Comment:
`resource` is boto3's concept of a default session that is shared globally, see: https://github.com/boto/boto3/blob/develop/boto3/__init__.py#L27-L34

But it's simply just creating an instance of `Session`, so we do this manually since we want to bind a `Session` per thread ourselves.